artemis/db.py

```diff
@@ -1,14 +1,34 @@
+import copy
 import dataclasses
 import datetime
+import functools
+import hashlib
+import json
 from enum import Enum
-from typing import Any, Dict, Generator, List, Optional, cast
+from typing import Any, Dict, Generator, List, Optional
 
 from karton.core import Task
 from pydantic import BaseModel
-from pymongo import ASCENDING, DESCENDING, MongoClient
+from sqlalchemy import (  # type: ignore
+    JSON,
+    Boolean,
+    Column,
+    Computed,
+    DateTime,
+    Index,
+    String,
+    create_engine,
+)
+from sqlalchemy.dialects.postgresql import TSVECTOR
+from sqlalchemy.dialects.postgresql import insert as postgres_upsert
```

Comment:
nah, leave it as `insert` because PSQL doesnt have `upsert` method

Reply:
right, done